packages/plugin-analytic-fb-pixel/src/FacebookPixelAnalytic.ts

```diff
@@ -0,0 +1,261 @@
+   * @override
+   * @param Page view data (containing path etc.).
+   * @returns TRUE when event has been hit; otherwise FALSE.
+   */
+  hitPageView(viewContentData: Record<string, any> | null = null) {
+    try {
+      if (!this.#fbq) {
+        throw new Error(
+          'Initialize the FacebookPixelHelper instance before calling hitPageView() method.'
+        );
+      } else if (typeof viewContentData !== 'object') {
+        throw new TypeError(
+          'Parameter data of hitPageView() method should be an object.'
+        );
+      }
+    } catch (error) {
+      this._processError(
+        error as Parameters<FacebookPixelAnalytic['_processError']>[0]
+      );
+
+      return false;
+    }
+
+    const hitResult = this.hit('PageView');
+
+    if (!hitResult) {
+      return false;
+    } else if (viewContentData) {
+      return this.hit('ViewContent', viewContentData);
+    }
+
+    return true;
+  }
+
+  /**
+   * Hits a search event (optionally with page name or other event data).
+   *
+   * @param Search query / event data.
+   * @param queryOrData
+   * @returns TRUE when event has been hit; otherwise FALSE.
+   */
+  hitSearch(queryOrData: Record<string, any> | string | null = null) {
+    try {
+      if (!this.#fbq) {
+        throw new Error(
+          'Initialize the FacebookPixelHelper instance before calling hitSearch() method.'
+        );
+      } else if (['string', 'object'].indexOf(typeof queryOrData) === -1) {
+        throw new TypeError(
+          'Parameter queryOrData of hitSearch() method should be a string or an object.'
+        );
+      }
+    } catch (error) {
+      this._processError(
+        error as Parameters<FacebookPixelAnalytic['_processError']>[0]
+      );
+
+      return false;
+    }
+
+    let eventData: Record<string, unknown> | null;
+
+    if (typeof queryOrData === 'string' && queryOrData) {
+      eventData = { search_string: queryOrData };
+    } else {
+      return this.hit('Search');
+    }
+
+    return this.hit('Search', eventData);
+  }
+
+  /**
+   * @override
+   * @inheritdoc
+   */
+  _configuration() {
+    // _configuration is only called on client, therefore window is defined
+    const clientWindow = this._window.getWindow()!;
+
+    if (
+      this.isEnabled() ||
+      !clientWindow[FB_ROOT_VARIABLE] ||
+      typeof clientWindow[FB_ROOT_VARIABLE] !== 'function'
+    ) {
+      return;
+    }
+
+    this._enable = true;
+
+    this.#fbq = clientWindow[FB_ROOT_VARIABLE];
+    this.#fbq!('init', this.getId());
+  }
+
+  /**
+   * @override
+   * @inheritdoc
+   */
+  _createGlobalDefinition(window: globalThis.Window) {
+    if (window[FB_ROOT_VARIABLE]) {
+      return;
+    }
+
+    const fbAnalytic = (window[FB_ROOT_VARIABLE] = function (
+      ...rest: unknown[]
+    ) {
+      fbAnalytic.callMethod
+        ? fbAnalytic.callMethod(...rest)
+        : fbAnalytic.queue.push(...rest);
+    }) as FbAnalyiticExtended;
+
+    if (!window[`_${FB_ROOT_VARIABLE}`]) {
```

Comment:
proč je potřeba mít nějaký "private" nakonfigurovaný def v global scope?

Reply:
bylo to tak implementováno už na začátku, aktuálně jsem tohle nevalidoval zda stále nutné => zůstane zachováno